tests/test_ipu_training.py

```diff
@@ -0,0 +1,41 @@
+import hydra
+import wandb
+import timeit
+
+from omegaconf import DictConfig, OmegaConf
+from loguru import logger
+from datetime import datetime
+from lightning.pytorch.utilities.model_summary import ModelSummary
+
+from graphium.config._loader import (
+    load_datamodule,
+    load_metrics,
+    load_architecture,
+    load_predictor,
+    load_trainer,
+    load_accelerator,
+    save_params_to_wandb,
+)
+from graphium.finetuning import modify_cfg_for_finetuning, GraphFinetuning
+from graphium.utils.safe_run import SafeRun
+
+from graphium.cli.train_finetune import run_training_finetuning
+import pytest
+
+FINETUNING_CONFIG_KEY = "finetuning"
+
+
+@pytest.mark.parametrize("acc_type", ["cpu", "ipu"])
+@pytest.mark.ipu
+def test_cli(acc_type) -> None:
+    """
+    The main CLI endpoint for training and fine-tuning Graphium models.
+    """
+    with hydra.initialize(version_base=None, config_path="../expts/hydra-configs"):
+        # config is relative to a module
+        cfg = hydra.compose(config_name="test", overrides=[f"accelerator={acc_type}"])
```

Comment:
Hey @s-maddrellmander, one more question I wanted to discuss: 

Do we need as many config files? 

Hydra supports a very elaborate [override syntax](https://hydra.cc/docs/advanced/override_grammar/basic/). I believe it will benefit the clarity and maintainability of the config system, if we actively try to minimize the number of configs laying around. Hydra automatically saves the full config to the [working/output directory](https://hydra.cc/docs/tutorials/basic/running_your_app/working_directory/#changing-or-disabling-hydras-output-subdir), so you can still easily save experiment-specific configs for reproducibility.

As for this specific case: I don't know for sure (please let me know what you think!), but I can imagine that we don't need 9 new configs, because we only want to override a few keys in the config (e.g. `constants.max_epochs` or `sample_size`). I now expect there to be a lot of repetition with the existing training configs. A simple example, instead of creating `expts/hydra-configs/test.yaml`, you could also do: 

```python
cfg = hydra.compose(config_name="test", overrides=[
    f"accelerator={acc_type}",
    "architecture=test",
    "task=test",
    "training=test",
])
```

I would prefer to have a long overrides list over many different new files. 

Let me know what you think!

Reply:
That's fair, still getting my head round all the hydra functionality, I'll look at adapting the existing configs via the override as much as possible. That'll be neater than having so many additional files to maintain.

Reply:
This is a lot better! Thank you for taking the time 🙏